### DISCUSSION TITLE
Opening to local network does not work

### DISCUSSION THREAD

**Discussion by Ken-vdE** [laravel/valet#1307] [Q&A]
### Description:
When following the steps given here: https://laravel.com/docs/9.x/valet#sharing-sites-on-your-local-network, sharing on my local network still does not work.

### Steps To Reproduce:
* https://laravel.com/docs/9.x/valet#sharing-sites-on-your-local-network
* Firewall disabled (MacOS)
* Go to URL using my local IP address as given by the example in the docs (in my case http://192.168.1.133/cool.test/). Also tried using other device on same network.
* Get a 404 page with `<title>Valet - Not Found<title>` only containing the words '404 - Not Found' (probably served by function show_valet_404() in server.php).
* I've also retried after updating everything:
    * brew upgrade
    * composer self-update
    * composer global update
    * valet install
    * edit valet.conf
    * valet restart
    * etc.

### Diagnosis
I think Valet's php script does not understand that URI with the IP address prefix.

**Comment by Ken-vdE** [comment]
I've played around with server.php but haven't gotten it to work.
These are my current code changes, maybe you can use it in some way.
Changed [line 113](https://github.com/laravel/valet/blob/3e5f16fee53db94233d16fd8f9f0e0b17fc98b09/server.php#L113)

```php
// https://github.com/laravel/valet/blob/3e5f16fee53db94233d16fd8f9f0e0b17fc98b09/server.php#L113
if ($_SERVER['HTTP_HOST'] === $_SERVER['SERVER_ADDR']) {
    $siteName = basename(
        explode('/', $_SERVER['REQUEST_URI'])[1] ?? null, 
        '.'.$valetConfig['tld']
    );
    $_SERVER['REQUEST_URI'] = substr($_SERVER['REQUEST_URI'], strlen($siteName . '.' . $valetConfig['tld']) + 1);
} else {
    $siteName = basename(
        // Filter host to support wildcard dns feature
        valet_support_wildcard_dns($_SERVER['HTTP_HOST'], $valetConfig),
        '.'.$valetConfig['tld']
    );
}
```

**Comment by dioncoda** [comment]
I was having the same problem on a clean install

I ran `valet restart` and now it's working

**Comment by Gnative** [comment]
This is what I've added to server.php file in `~/.composer/vendor/laravel/valet/server.php` around line 104
This allows address such is as outlined in [https://laravel.com/docs/9.x/valet#sharing-sites-on-your-local-network](https://laravel.com/docs/9.x/valet#sharing-sites-on-your-local-network)

```php
/**
* If the HTTP_HOST is an IP address, check the start of the REQUEST_URI for a
* valid hostname, extract and use it as the effective HTTP_HOST in place
* of the IP. It enables the use of Valet in a local network.
*/
if (preg_match('/^([0-9]+\.){3}[0-9]+$/', $_SERVER['HTTP_HOST'])) {
  $uri = ltrim($_SERVER['REQUEST_URI'], '/');

  if (preg_match('/^[-.0-9a-zA-Z]+\.'. $valetConfig['tld'] .'/', $uri, $matches)) {
     $host = $matches[0];
     $_SERVER['HTTP_HOST'] = $host;
     $_SERVER['REQUEST_URI'] = str_replace($host, '', $uri);
  }
}
```

**Comment by Gnative** [comment]
There is a pull request from @thinkverse for this  that been hanging around for a while - https://github.com/laravel/valet/pull/1284/files

**Comment by Small-Systems** [comment]
There can not only be there four of us facing this problem. Has anyone come up with a reliable solution? The code above doesn't work for me.
